SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,680 @@
+﻿namespace SMLHelper.V2.Options
+{
+    using HarmonyLib;
+    using QModManager.API;
+    using SMLHelper.V2.Interfaces;
+    using SMLHelper.V2.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using UnityEngine;
+    using UWE;
+    using Logger = V2.Logger;
+#if SUBNAUTICA
+    using Text = UnityEngine.UI.Text;
+#elif BELOWZERO
+    using Text = TMPro.TextMeshProUGUI;
+#endif
+
+    /// <summary>
+    /// An internal derivative of <see cref="ModOptions"/> for use in auto-generating a menu based on attributes
+    /// declared in a <see cref="ConfigFile"/>.
+    /// </summary>
+    /// <typeparam name="TConfigFile">The type of the class derived from <see cref="ConfigFile"/> to use for
+    /// loading to/saving from disk.</typeparam>
+    internal class ConfigModOptions<TConfigFile> : ModOptions where TConfigFile : ConfigFile, new()
```

Comment:
This file is pretty big. Consider using `#region`

Reply:
I've added a couple of regions for grouping methods used for similar things together. What do you think?